sched/queue/memory/simple.go

```diff
@@ -48,33 +142,28 @@ func (q *simpleQueue) Enqueue(job sched.Job) (string, error) {
 		return "", err
 	}
 	resultCh := make(chan enqueueResult)
-	q.inCh <- enqueueReq{job, resultCh}
+	q.reqCh <- enqueueReq{job, resultCh}
 	result := <-resultCh
 	return result.jobID, result.err
 }
 
-func (q *simpleQueue) Close() error {
-	close(q.inCh)
-	close(q.outCh)
-	return nil
+type statusResult struct {
+	status sched.JobStatus
+	err    error
 }
 
-func (q *simpleQueue) loop() {
-	for {
-		req, ok := <-q.inCh
-		if !ok {
-			// Channel is cloed
-			return
-		}
-		job := req.job
-		id := fmt.Sprintf("%v", q.nextID)
-		q.nextID++
-		job.Id = id
-		select {
-		case q.outCh <- simpleWorkItem(job):
-			req.resultCh <- enqueueResult{id, nil}
-		default:
-			req.resultCh <- enqueueResult{"", queue.NewCanNotScheduleNow(1*time.Second, "queue full")}
-		}
-	}
+func (q *simpleQueue) Status(jobId string) (sched.JobStatus, error) {
+	resultCh := make(chan statusResult)
+	q.reqCh <- statusReq{jobId, resultCh}
+	result := <-resultCh
```

Comment:
I've been talking to Dan about the general design here and he's following the ''share by communicating' model whereas I worked up an alternative implementation which relies on locking shared data. So the answer to your question is that it's avoiding locking.